MetadataExtractor/IO/BufferReader.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Buffers.Binary;
+
+namespace MetadataExtractor.IO;
+
+internal ref struct BufferReader(ReadOnlySpan<byte> bytes, bool isBigEndian)
+{
+    private readonly ReadOnlySpan<byte> _bytes = bytes;
+    private int _position = 0;
+    private bool _isBigEndian = isBigEndian;
+
+    public long Position => _position;
+
+    public byte GetByte()
+    {
+        if (_position >= _bytes.Length)
+            throw new IOException("End of data reached.");
+
+        return _bytes[_position++];
+    }
+
+    public void GetBytes(scoped Span<byte> bytes)
+    {
+        if (_position + bytes.Length > _bytes.Length)
+            throw new IOException("End of data reached.");
+
+        _bytes.Slice(_position, bytes.Length).CopyTo(bytes);
+        _position += bytes.Length;
+    }
+
+    public byte[] GetBytes(int count)
+    {
+        if (_position + count > _bytes.Length)
+            throw new IOException("End of data reached.");
+
+        var bytes = new byte[count];
+
+        _bytes.Slice(0, count).CopyTo(bytes);
```

Comment:
Is the zero correct here?

Reply:
NO! ERG. 

I'll extend our test coverage to make this path is hit.